src/common/logger/logger.cpp

```diff
@@ -33,31 +33,25 @@ level::level_enum getLogLevel(std::wstring_view logSettingsPath)
     return result;
 }
 
-Logger::Logger()
-{
-}
+std::shared_ptr<spdlog::logger> Logger::logger;
 
-Logger::Logger(std::string loggerName, std::wstring logFilePath, std::wstring_view logSettingsPath)
+void Logger::init(std::string loggerName, std::wstring logFilePath, std::wstring_view logSettingsPath)
 {
     auto logLevel = getLogLevel(logSettingsPath);
     try
     {
         auto sink = make_shared<sinks::daily_file_sink_mt>(logFilePath, 0, 0, false, LogSettings::retention);
-        this->logger = make_shared<spdlog::logger>(loggerName, sink);
+        logger = make_shared<spdlog::logger>(loggerName, sink);
     }
     catch (...)
     {
         cerr << "Can not create file logger. Create stdout logger instead" << endl;
-        this->logger = spdlog::stdout_color_mt("some_unique_name");
+        logger = spdlog::stdout_color_mt(loggerName);
```

Comment:
let's not silently start logging to the stdout, and instead show a standard message box error dialog about the logger problem. at the same time, since we must have a non-null `logger` object, we could assign `spdlog::null_logger_mt("null");` to it in the `catch` stmt.

Reply:
done